File: frontend/src/api/social.js
import http from './http'

export async function getFollowStatus(accountId) {
  const response = await http.get('/social/status', {
    params: { vlogger_id: String(accountId) },
  })

  return response.data
}

export async function followAccount(accountId) {
  const response = await http.post('/social/follow', {
    vlogger_id: String(accountId),
  })

  return response.data
}

export async function unfollowAccount(accountId) {
  const response = await http.post('/social/unfollow', {
    vlogger_id: String(accountId),
  })

  return response.data
}

export async function getFollowingAccounts({ cursor = '', limit = 12 } = {}) {
  const params = { limit }

  if (cursor) {
    params.cursor = cursor
  }

  const response = await http.get('/social/following', { params })
  return response.data
}

export async function getFollowerAccounts({ cursor = '', limit = 12 } = {}) {
  const params = { limit }

  if (cursor) {
    params.cursor = cursor
  }

  const response = await http.get('/social/followers', { params })
  return response.data
}

File: backend/router/social.go
package router

import (
	"video_feedsystem/handler"
	"video_feedsystem/middleware"

	"github.com/cloudwego/hertz/pkg/app/server"
)

// registerSocialRoutes 注册关注状态、关注操作和关注关系列表接口。
func registerSocialRoutes(h *server.Hertz) {
	social := h.Group("/social")
	{
		authorized := social.Group("", middleware.JWTAuth())
		{
			authorized.POST("/follow", handler.FollowUser)
			authorized.POST("/unfollow", handler.UnfollowUser)
			authorized.GET("/status", handler.GetFollowStatus)
			authorized.GET("/following", handler.GetFollowingList)
			authorized.GET("/followers", handler.GetFollowerList)
		}
	}
}
